fix: Allow fetching of dynamic items on select (#2535)
Workday/canvas-kit: modules/react/select/stories/Select.stories.mdx

<ExampleCodeBlock code={Error} />

### Initial Selected Item

You can set `initialSelectedIds` to the value that you want initially selected.

<ExampleCodeBlock code={InitialSelectedItem} />

### Placeholder

You can change the placeholder text by passing in a string value to the `placeholder` attribute on
the `Select.Input`.

<ExampleCodeBlock code={Placeholder} />

### Fetching Dynamic Items

It's common to load items from a server call. Hoisting the `model` and setting your items on state

> allows you to pass those items to your `model`. You can leverage React State to set your items on

Reviewer: ```suggestion
allows you to pass those items to your `model`. You can leverage React state to set your items on
```

load as well as displaying a placeholder indicating when items are loaded.

**Note: In this case we need to use `getId` and `getTextValue` because our data doesn't have the
properties of `id` or `text`. Using these helper functions sets the `serverId` to be `id` and
`label` to be `text`.**

<ExampleCodeBlock code={FetchingDynamicItems} />

### Complex

> When registering items in an array of objects, it's common to have the text displayed to the user be

Reviewer 2: ```suggestion
When registering items in an array of objects, it's common to have the text that is displayed to the user be
```

> different than an id. In this example, `serverId` and `label` properties need to be re mapped to

Reviewer 2: ```suggestion
different than an id. In this example, `serverId` and `label` properties need to be remapped to
```

`id` and `text` hence the usage of `getId` and `getTextValue`. If your object has the properties
`text` and `id`, there would be no need for this.

<ExampleCodeBlock code={Complex} />

> **Note:By default, the identifier and text value are `id` and `text` properties respectively. If

Reviewer: ```suggestion
**Note: By default, the identifier and text value are `id` and `text` properties respectively. If
```

</Select>;
```

### When to use `getId`, or `getTextValue`

> - `getId`: This is an optional function to return an id of an item. If not provided, the default

Reviewer: ```suggestion
- `getId`: This is an optional function to return the id of an item. If not provided, the default
```

  function will return the `id` property from the object of each item. If you did not provide

>   `items`, do not override this function. If you don't provided `items` and instead provide static

Reviewer: ```suggestion
  `items`, do not override this function. Instead provide static
```

>   items via JSX, the list will create an internal array of items where `id` is the only property and

Reviewer: ```suggestion
  items via JSX. the list will create an internal array of items where `id` is the only property and
```

  the default `getId` will return the desired result. **Note: If your array of objects has a
  different property for `id`, like `serverId`, use this function to set the id.**

  ```tsx

>   const options = [{text: 'Pizza', serverId: 'pizza-1'}, {text: 'CheeseBurger', serverId: 'cheese-burger'}]

Reviewer: ```suggestion
  const options = [{text: 'Pizza', serverId: 'pizza-1'}, {text: 'Cheeseburger', serverId: 'cheeseburger'}]
```

  <Select items={options} getId={(item) => item.serverId}>
    <FormField label="Your Label">
      <Select.Input onChange={e => handleChange(e)} id="contact-select" />
      <Select.Popper>
        <Select.Card>
          <Select.List>{item => <Select.Item>{item.text}</Select.Item>}</Select.List>
        </Select.Card>
      </Select.Popper>
    </FormField>
  </Select>
  ```

- `getTextValue`: Optional function to return the text representation of an item. If not provided,
  the default function will return the `text` property of the object of each item or an empty string
  if there is no `text` property. If you did not provide `items`, do not override this function.
  **Note: If your array of objects has a different property for `text`, like `label`, use this
  function to set the text.**

  ```tsx

>   const options = [{label: 'Pizza', id: 'pizza-1'}, {label: 'CheeseBurger', id: 'cheese-burger'}]

Reviewer: ```suggestion
  const options = [{label: 'Pizza', id: 'pizza-1'}, {label: 'Cheeseburger', id: 'cheeseburger'}]
```
